UPGRADE.md

```diff
@@ -1939,6 +1939,131 @@ If you've declared custom Record types, you'll need to upgrade your code as foll
+
+To upgrade simply replace all occurrences of `FormWithRedirect` with  `Form`.
+
+```diff
+- import { FormWithRedirect } from 'react-admin';
++ import { Form } from 'react-admin';
+
+export const MyForm = () => (
+-    <FormWithRedirect
++    <Form
+        render={() => ...}
+    />
+);
+```
+
+### `<FormWithRedirect>` Props Have Changed
+
+`<FormWithRedirect>` used to accept [`react-final-form` `<Form>` props](https://final-form.org/docs/react-final-form/types/FormProps). It now accepts [`react-hook-form` `useForm` props](https://react-hook-form.com/api/useform). This also affects the other form components (`SimpleForm`, `TabbedForm`, etc.)
+
+The most commonly used prop is probably `initialValues` which is now named `defaultValues`:
+
+```diff
+const PostCreate = () => (
+    <Create>
+        <SimpleForm
+-            initialValues={{ title: 'A default title' }}
++            defaultValues={{ title: 'A default title' }}
+        >
+            ...
+        </SimpleForm>
+    </Create>
+)
+```
+
+However, we kept the `validate` function prop which we automatically translate as a custom [`react-hook-form` `resolver`](https://react-hook-form.com/api/useform#validationResolver).
+
+This also means you can now use [`yup`](https://github.com/jquense/yup), [`zod`](https://github.com/colinhacks/zod), [`joi`](https://github.com/sideway/joi), [superstruct](https://github.com/ianstormtaylor/superstruct), [vest](https://github.com/ealush/vest) or any [resolvers](https://react-hook-form.com/api/useform#validationResolver) supported by `react-hook-form` to apply schema validation.
+
+### `<FormWithRedirect>` Render Function Arguments Have Changed
+
+The `<FormWithRedirect>` render function used to be called with an object containing parts of the `final-form` form state (`valid`, `invalid`, `pristine`, `dirty`). It now only receive the `handleSubmit` function which must be passed to the `onSubmit` prop of the underlying form. If you need access to the form state, call [`react-hook-form` `useFormState` hook](https://react-hook-form.com/api/useformstate):
+
+```diff
+import { FormWithRedirect } from 'react-admin';
++ import { useFormState } from 'react-hook-form';
+
+const MyCustomForm = () => {
+    return (
+-        <FormWithRedirect
++        <Form
+-            render={({ valid, dirty, handleSubmit }) => (
++            render={({ handleSubmit }) => (
+                <form onSubmit={handleSubmit}>
+                    ...
+-                    <SubmitButton disabled={!dirty || !valid}>Save</SubmitButton>
++                    <SubmitButton>Save</SubmitButton>
+                </form>
+            )}
+        />
+    );
+};
+
+-const SubmitButton = ({ disabled, ...props }) => {
++const SubmitButton = (props) => {
++    const { isDirty, isValid } = useFormState();
+    return (
+-        <button disabled={disabled} {...props} />
++        <button disabled={!isDirty || !isValid} {...props} />
+    );
+}
+```
+
+### `useInput` Signature Changed
```

Comment:
You're listing changes to all Input components. Maybe the title of the section should be changed.